File: src/main/java/org/cytoscape/biogwplugin/internal/query/QueryParameter.java
package org.cytoscape.biogwplugin.internal.query;

import java.util.HashMap;

public class QueryParameter {
	
	public enum ParameterType {
		TEXT, CHECKBOX, COMBOBOX, UNIPROT_ID, ONTOLOGY, OPTIONAL_URI
	}
	
//	public class Option {
//		String name;
//		String value;
//		
//		public Option(String name, String value) {
//			super();
//			this.name = name;
//			this.value = value;
//		}
//		
//	}
	
	String id;
	String name;
	ParameterType type;
	//ArrayList<Option> options;
	String value = null;
	HashMap<String, String> options;
	
	public QueryParameter(String id, String name, ParameterType type) {
		super();
		this.id = id;
		this.name = name;
		this.type = type;
		this.options = new HashMap<>();
	}
	
	public void addOption(String name, String value) {
		options.put(name, value);
	}
}

File: src/main/java/org/cytoscape/biogwplugin/internal/query/QueryParameter.kt
package org.cytoscape.biogwplugin.internal.query

import java.util.HashMap

class QueryParameter(var id: String, var name: String, var type: QueryParameter.ParameterType) {

    enum class ParameterType {
        TEXT, CHECKBOX, COMBOBOX, UNIPROT_ID, ONTOLOGY, OPTIONAL_URI
    }

    class EnabledDependency(val dependingParameter: String, val isEnabled: Boolean, val forParameterValue: String)

    var value: String? = null
    var dependency: EnabledDependency? = null
    var options: HashMap<String, String> = HashMap<String, String>()

    fun addOption(name: String, value: String) {
        options.put(name, value)
    }
}
